### DISCUSSION TITLE
Data pipeline using DVC

### DISCUSSION THREAD

**Discussion by ijlyttle** [quarto-dev/quarto-cli#6805] [Show and Tell]
### Demo of DVC with Quarto

[DVC](https://dvc.org/) is an open-source project, an acronym for "data version control". It can be used to:

  - specify the dependencies and outputs of a pipeline
  - manage remote-storage of data

This sounds a lot like [targets](https://books.ropensci.org/targets/) (for R) and pins (for [R](https://pins.rstudio.com/) and [Python](https://rstudio.github.io/pins-python)), both of which I respect greatly. One of the big differences is that DVC can live outside your code, as an orchestrator, whereas, my understanding of targets and pins is that they are integrated into the pipeline code.

As an orchestrator, I wanted to try out DVC as a complementary tool to Quarto, so I made a small [pipeline](https://ijlyttle.github.io/grid-france/) ([repo](https://github.com/ijlyttle/grid-france)). This pipeline is made using Python, but like Quarto, I don't think DVC "cares" about the languages used in the pipeline; it just runs `quarto render` on a file if it is invalidated. This required me to `freeze` the Quarto files in the pipeline, so that a global `quarto render` will work for the "rest" of the project.

I use an S3 bucket as backing, but DVC supports a variety of remote storage.

One point that I wanted to bring up here is that I found `.qmd` files to be suitable for the pipeline, but not `.ipynb` files. I may be putting this imprecisely, but running a `.qmd` file does not change the `.qmd` file; however, running a `.ipynb` file changes it. If a pipeline's dependencies includes a `.ipynb` file, I can see things becoming circular.

That said, there is an [issue at DVC](https://github.com/iterative/dvc/issues/3439#issuecomment-663873682) where they discuss allowing a dependency to filter for the (in this case) Python code - but it seems to have stalled.

FWIW, I'll be at posit::conf() if anyone wants to discuss. Thanks!

**Comment by cderv** [comment]
Thanks for sharing  @ijlyttle  ! 

DVC is indeed an interesting tool ! I won't be physically at conf, but happy to discuss all this virtually or next time we'll meet!

**Reply by cderv** [reply]
And nice project on French Data ! I also used to work at RTE so really nice project ! Glad public data are useful ! 😉

**Reply by ijlyttle** [reply]
Thanks, @cderv - I had forgotten that you had worked at RTE 😳 - I appreciate how open they make things, hoping I can do something useful with it beyond this pipeline!

Too bad you will not be physically at conf - but I will take you up on your offer to discuss virtually! (maybe a little after conf?)

**Reply by cscheid** [reply]
@ijlyttle There'll be a "posit::conf() lounge", and quarto folks will host a meet-and-greet session on Tuesday, September 19⋅2:20 – 4:00pm CDT. It'd be great to see you!

**Reply by ijlyttle** [reply]
@cscheid Super, I'll be there!

**Comment by DaveParr** [comment]
This sounds neat. Hope the talk went well. Do you have any demo code or presentation published on your session? Would love to have a closer look at dvc cml + quarto.
